zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/GlobalControllerExceptionHandler.java

```diff
@@ -33,7 +33,8 @@ public class GlobalControllerExceptionHandler extends ResponseEntityExceptionHan
       "Required request body is missing";
   private static final String REQUEST_BODY_PARSE_EXCEPTION_MESSAGE =
       "Request property [%s] cannot be parsed";
-  private static final String INVALID_ENUM_VALUE_EXCEPTION_MESSAGE = "Invalid Enum value";
+  // Thrown by the OpenApi in case of an invalid enum value
+  private static final String INVALID_ENUM_VALUE_EXCEPTION_MESSAGE = "Unexpected value";
```

Comment:
💭 Two ideas:
1. Can we be more explicit here in what is unexpected? Is the enum class and/or the field (key in the JSON) known? This could help create a more useful message for users, e.g. "The value provided in '<field>' is unsupported".
2. If possible, but this is an absolute bonus: if we know the enum class (because it's in the error message), we can even fetch and return the possible values, e.g. "The value provided in '<field>' is unsupported, it must be any of [<list of enum values>]".

Reply:
> If possible, but this is an absolute bonus: if we know the enum class (because it's in the error message), we can even fetch and return the possible values, e.g. "The value provided in '' is unsupported, it must be any of [<list of enum values>]".

I have used the `INVALID_ENUM_VALUE_EXCEPTION_MESSAGE ` from OpenApiSPEC to validate the message. It comes very generic and we have a lot of ENUMS. 
The OpenAPI Spec does not bring the possible Enum values over this case. I think this would be not 

> 1. Can we be more explicit here in what is unexpected? Is the enum class and/or the field (key in the JSON) known? This could help create a more useful message for users, e.g. "The value provided in '<field>' is unsupported".

I can change in the line down there, this constant is only use to validate the error caused by the Open API Spec, I can update the error message over here: 
`detail = Objects.requireNonNull(httpMessageNotReadableException.getRootCause()).getMessage();`
And update the message accordingly.

Reply:
To give a better example, this is the stack trace from the Open API Spec from ENUMS

`org.springframework.http.converter.HttpMessageNotReadableException: JSON parse error: Cannot construct instance of `io.camunda.zeebe.gateway.protocol.rest.SortOrderEnum`, problem: Unexpected value 'dsc'
`

Reply:
I gave this a quick look and discovered the following:
* The `HttpMessageNotReadableException` we receive has a cause exception of `ValueInstantiationException`.
* The value instantiation exception can tell you the location of the issue in the source JSON, i.e., the field where it occurs.

With that, we should be able to give a better hint here where the problematic field is. We could even use the defined type information (`io.camunda.zeebe.gateway.protocol.rest.SortOrderEnum`) to fetch all possible values and return those to the user.

For example, with the following criteria we can check if we have an enum issue:
```
  private boolean isUnknownEnumError(final Exception ex) {
    return ex instanceof HttpMessageNotReadableException
        && ex.getCause() instanceof final ValueInstantiationException instantiationException
        && instantiationException.getType().isEnumType();
  }
```

The following code would get the details we need:
```
...
    } else if (isUnknownEnumError(ex)) {
      final var instantiationException = (ValueInstantiationException) ex.getCause();
      final var options =
          Arrays.stream(instantiationException.getType().getRawClass().getEnumConstants())
              .map(Objects::toString)
              .toList();
      final var field = instantiationException.getPath().getLast().getFieldName();
      detail =
          "%s for enum field '%s'. Use any of the following values: %s"
              .formatted(
                  ((HttpMessageNotReadableException) ex).getRootCause().getMessage(),
                  field,
                  options);
    } else {
...
```

This would lead to the following error message for the enum:
```
Unexpected value 'dsc' for enum field 'order'. Use any of the following values: [ASC, DESC]
```

Reply:
Thanks for driving this @tmetzke , this works. 
Super interesting, I miss this kind of exception (to be honest pretty new for me), I was checking around the `InvalidArgumentException`